tooling/maven/camel-maven-plugin/src/main/java/org/apache/camel/maven/RouteCoverageMojo.java

```diff
@@ -348,10 +441,53 @@ private String templateCoverageData(String fileName, String routeId, List<RouteC
         double percentage = ((double) covered / (double) model.size()) * 100;
         sw.println();
         sw.println("Coverage: " + covered + " out of " + model.size() + " (" + String.format("%.1f", percentage) + "%)");
-        sw.println();
-
+              
         return bos.toString();
+    }    
+    
+    private Attr createAttrInt(Document doc, Element e, String name, Integer value) {
+    	Attr a = doc.createAttribute(name);
+    	a.setValue(value.toString());
+    	e.setAttributeNode(a);
+    	
+    	return a;
+    }
+    
+    private Attr createAttrString(Document doc, Element e, String name, String value) {
+    	Attr a = doc.createAttribute(name);
+    	a.setValue(value);
+    	e.setAttributeNode(a);
+    	
+    	return a;
+    }
+    
+    private Document createDocument() {
+    	Document document = null;
+    	try {	
+    	    DocumentBuilderFactory documentFactory = DocumentBuilderFactory.newInstance();        
+    	    DocumentBuilder documentBuilder = documentFactory.newDocumentBuilder();
+    	    document = documentBuilder.newDocument();
+        
+        } catch (ParserConfigurationException pce) {
+            pce.printStackTrace();
```

Comment:
dont do this but let the exception be propagated  back up so we can have a better error reporting about not possible to generate jacoco report due to ....

Also these methods can be static methods